students/src/main/java/com/example/students/service/StudentService.java

```diff
@@ -15,5 +15,51 @@
 @Service
 @RequiredArgsConstructor
 public class StudentService {
+    private final StudentRepository studentRepository;
+    private final StudentClient client;
+    StudentMapper studentMapper = new StudentMapper();
+    public StudentFullName getStudentFullName (Integer id) {
+        RestTemplate restTemplate = new RestTemplate();
+        StudentFullName fullName = new StudentFullName();
 
+        String resourceURL = String.format("http://name:8085/name/%d", id);// service: first-name
+        final String firstName = restTemplate.getForObject(resourceURL, String.class);
+        fullName.setFirstName(firstName);
+
+        if(studentRepository.findById(id).isPresent()){
+            fullName.setSecondName(studentRepository.findById(id).get().getSecondName());
+        }
+
+        fullName.setLastName(client.getStudentLastName(id).getLastName());//service: last-name
+        return fullName;
+    }
+    public StudentDTO createStudent(StudentDTO studentDTO){ //move to another method
+
+        return studentMapper
+                .dtoMapper(studentRepository
+                                .save(studentMapper
+                                                .entityMapper(studentDTO)
+                                )
+                );
+    }
+    public List<StudentEntity> getAllStudents(){
+        return studentRepository.findAll();
+    }
+    public Optional<StudentEntity> getStudentById(Integer id){
+        return studentRepository.findById(id);
+    }
+    public Integer deleteStudentById (Integer id){
+        studentRepository.deleteById(id);
+        return  id;
+    }
+    public StudentDTO updateStudent (Integer id, StudentDTO studentDTO){
+            StudentEntity studentEntity = studentRepository.getReferenceById(id);
```

Comment:
maybe should be like findById or getById

Reply:
***getReferenceById(id)*** is a internal method SpringBoot

Reply:
`getById()` on documentation marked as deprecated and replaced `getReferenceById()`